### PR TITLE
RavenDB-21983 - Update our Client API XML documentation - Shiran

### PR DESCRIPTION
**Pull request by shiranshalom**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21983/Update-our-Client-API-XML-documentation-Shiran

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs [hunk 1]
```diff
@@ -15,20 +15,23 @@ namespace Raven.Client.Documents.Session
     /// </summary>
     public partial class AsyncDocumentSession
     {
+        /// <inheritdoc cref="IAsyncDocumentSession.TimeSeriesFor"/>
```
Comment (garayx) [inline src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs]: shouldnt this be
```suggestion
        /// <inheritdoc cref="IAsyncDocumentSession.TimeSeriesFor(string, string)"/>
```

File: src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs [hunk 2]
```diff
@@ -15,20 +15,23 @@ namespace Raven.Client.Documents.Session
...
 
+        /// <inheritdoc cref="IAsyncDocumentSession.TimeSeriesFor(object, string)"/>
         public IAsyncSessionDocumentTimeSeries TimeSeriesFor(object entity, string name)
         {
             ValidateTimeSeriesName(name);
 
             return new AsyncSessionDocumentTimeSeries<TimeSeriesEntry>(this, entity, name);
         }
 
+        /// <inheritdoc cref="IAsyncDocumentSession.TimeSeriesFor{TValue}"/>
```
Comment (garayx) [inline src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs]: shouldnt this be


```suggestion
        /// <inheritdoc cref="IAsyncDocumentSession.TimeSeriesFor{TValue}(string, string)"/>
```

File: src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs [hunk 3]
```diff
@@ -446,6 +446,7 @@ private Task<TimeSeriesEntry<TValues>[]> GetAsyncInternal(DateTime? from, DateTi
             return GetTypedFromCache<TValues>(from, to, includes: null, start, pageSize, token);
         }
 
+        /// <inheritdoc cref="IAsyncSessionDocumentTypedTimeSeries{TValue}.GetAsync"/>
```
Comment (garayx) [inline src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs:449]: do we want to add the token as well?

File: src/Raven.Client/Documents/Session/IAsyncAdvancedSessionOperations.Stream.cs [hunk 4]
```diff
@@ -27,49 +27,67 @@ public partial interface IAsyncAdvancedSessionOperations :
     {
 
     }
-    
+
     public interface IAsyncTimeSeriesSessionStreamOperations<T>
     {
+        /// <inheritdoc cref="IDocumentsSessionOperations.Stream{T}(IQueryable{T}) "/>
```
Comment (garayx) [inline src/Raven.Client/Documents/Session/IAsyncAdvancedSessionOperations.Stream.cs]: dont we need to add the token for all the methods below?

File: src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs (hunk 1 above)
Comment (shiranshalom) [reply src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs]: done

File: src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs (hunk 2 above)
Comment (shiranshalom) [reply src/Raven.Client/Documents/Session/AsyncDocumentSession.TimeSeriesFor.cs]: done

File: src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs (hunk 3 above)
Comment (shiranshalom) [reply src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs:449]: it's there :)

File: src/Raven.Client/Documents/Session/IAsyncAdvancedSessionOperations.Stream.cs (hunk 4 above)
Comment (shiranshalom) [reply src/Raven.Client/Documents/Session/IAsyncAdvancedSessionOperations.Stream.cs]: done

**Pull request merged**